packages/sdk/src/ZSshUtils.ts

```diff
@@ -50,18 +50,43 @@ export class ZSshUtils {
         };
     }
 
-    public static async installServer(session: SshSession, serverPath: string, localDir: string): Promise<void> {
+    public static async installServer(
+        session: SshSession,
+        serverPath: string,
+        localDir: string,
+        onProgress?: (increment: number) => void, // Callback to report incremental progress
```

Comment:
4 parameters I think it's fine, but if we need to add a new one, maybe then we move the optional `onProgress` inside an options object.